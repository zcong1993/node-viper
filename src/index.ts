export class Viper {
  constructor() {}
}

import * as path from 'path'

console.log(path.resolve('$HOME/zcong'))
