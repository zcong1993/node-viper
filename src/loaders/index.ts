export * from './dotEnv'
export * from './env'
export * from './json'
export * from './toml'
export * from './yaml'
