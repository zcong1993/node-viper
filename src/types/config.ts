export type supportTypes = 'json' | 'yaml' | 'toml' | 'dotenv'

export interface RootConfig {
  configName?: string
  configPaths?: string[]
  configTypes?: supportTypes[]
  envPrefix?: string
}
