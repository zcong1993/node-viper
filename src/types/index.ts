export * from './config'

export interface Loader {
  load(): object | null
}
