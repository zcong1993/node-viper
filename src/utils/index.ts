import { resolve, sep } from 'path'
import { readFileSync } from 'fs'
import { homedir } from 'os'

export const absPathify = (p: string): string => {
  let pp: string = p

  if (pp.startsWith('$HOME')) {
    pp = homedir() + p.slice(5)
  }

  if (pp.startsWith('$')) {
    const index = pp.indexOf(sep)
    pp = process.env[p.slice(1, index)] + p.slice(index)
  }

  return resolve(pp)
}

export const readFile = (p: string, encode: string = 'utf8') =>
  readFileSync(p, encode)
