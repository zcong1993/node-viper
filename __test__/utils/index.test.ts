import { absPathify, normalizeConfig, getExt } from '../../src/utils'
import { RootConfig, supportTypes } from '../../src/types'

it('absPathify should works well', () => {
  expect(absPathify('$HOME/test').startsWith('$HOME')).toBeFalsy()
  expect(absPathify('$PWD/test')).toBe(process.cwd() + '/test')
  expect(absPathify('.')).toBe(process.cwd())
})

it('normalizeConfig should works well', () => {
  const configs: RootConfig[] = [
    {},
    {
      configName: 'test',
      configPaths: ['.']
    },
    {
      configTypes: ['json']
    }
  ]
  configs.forEach(config => expect(normalizeConfig(config)).toMatchSnapshot())
})

it('should throw for bad config', () => {
  const configs: RootConfig[] = [
    {
      configName: 'test'
    },
    {
      configPaths: ['.']
    }
  ]

  configs.forEach(config => {
    expect(() => normalizeConfig(config)).toThrow()
  })
})

it('getExt should works well', () => {
  expect(getExt('dotenv')).toEqual([''])
  expect(getExt('json')).toEqual(['.json'])
  expect(getExt('toml')).toEqual(['.toml'])
  expect(getExt('yaml')).toEqual(['.yml', '.yaml'])
})
