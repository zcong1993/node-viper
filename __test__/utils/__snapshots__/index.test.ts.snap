// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeConfig should works well 1`] = `
Object {
  "configTypes": Array [
    "json",
    "yaml",
    "toml",
    "dotenv",
  ],
}
`;

exports[`normalizeConfig should works well 2`] = `
Object {
  "configName": "test",
  "configPaths": Array [
    ".",
  ],
  "configTypes": Array [
    "json",
    "yaml",
    "toml",
    "dotenv",
  ],
}
`;

exports[`normalizeConfig should works well 3`] = `
Object {
  "configTypes": Array [
    "json",
  ],
}
`;
